### DISCUSSION TITLE
Require supply to be issued in first numeric issuance

### DISCUSSION THREAD

**Discussion by jdogresorg** [CounterpartyXCP/CIPs#91] [CIPs]
I’ve been thinking about ways to limit what I consider “abuse” of numeric assets. 

Named assets and Subassets pay a small amount of XCP as an anti-spam mechanism. Numeric assets have no such XCP anti-spam protections, so I feel additional rules and restrictions could/should be put in place to ensure responsible asset usage and limit “abuse” of the system. 

To use the features of CP, one must issue some supply in order to put said supply in orders, dispensers, sends, etc.

Issuing numeric tokens with no supply, and token supply locked, means the token will never be able to use the features of CP.

Perhaps we should update numeric asset issuances to require supply be issued in the first issuance. This would, ensure that numeric tokens are only being used in a way where there is at least the potential for use of the CP features. I think this would also discourage issuing numeric assets unless they are actually going to be using the CP platform features.

This change would mean no numeric asset name squatting/pre-registrations, and no issuing 0 supply tokens that can’t be used on CP

**Comment by jotapea** [comment]
What prevents these issuers from specifying a quantity 1 (or any number) instead of 0, and then continue business as usual?

I still believe the best way to proceed is to add a fee to numerics. All Counterparty assets must pay an XCP fee to be registered.

**Reply by jdogresorg** [reply]
Nothing prevents users from just specifying a quantity of 1, however at least with 1 supply issued, there is the possibility of using the CP features. 

With no supply issued, no way to use CP.

I think longer term a fee on numerics will be necessary, but with some simple changes like this, hopefully we can guide users how to use the numerics in a responsible way, and keep the “free” numerics around for a bit longer.

**Reply by Jpja** [reply]
How about disallowing locking at 0 supply in general?

This might kill some potential use-cases (use of ownership as a dynamic asset where current holder can update the metadata, in contrast to a static token) but if such usage is harmful to CP (ie bloating the issuances table), it might be a good proactive solution to remove the possibility of 0-issuance locked assets.

Although, as @jotapea points out, nothing prevents issuing 1 token instead. It can be burned. All we can do is add some friction, so maybe an XCP issuance fee is the better solution overall?

**Reply by jdogresorg** [reply]
Preventing locking with 0 supply issued also might make sense.

**Reply by jotapea** [reply]
There can be creative use cases where users might want to zero lock issuances. These assets can still be transferred and issue description updates, so I wouldn't say that they don't use Counterparty features. Adding arbitrary limitations that just to add some easily bypassed friction is not a real solution to the problem.

**Comment by adamkrellenstein** [comment]
I think if there's a use case that a lot of people have for which they're issuing a locked 0-supply token, then we should consider creating a new transaction type that's tailored for them. Then we could avoid filling up the issuances with fake-issuances, and optimize the implementation appropriately. (For instance, we could optionally prune data the way Bitcoin prunes `OP_RETURN`.)

*Then* we could block null, locked issuances, which don't make any sense. :)

**Comment by jotapea** [comment]
> I still believe the best way to proceed is to add a fee to numerics. All Counterparty assets must pay an XCP fee to be registered.

For [posterity](https://github.com/CounterpartyXCP/cips/issues/77#issuecomment-1796397979), I no longer think like this.

And @adamkrellenstein , I think that the other transaction type that is already available is `broadcasts`. And I think the main difference compared to 0 locked `issuances` is the possibility of issuance transfer, which I believe is a legit reason to choose it over a broadcast. And named assets that are 0 locked have clear use cases as "superassets" (ability to create sub-assets).

If the database schema is [optimized](https://github.com/CounterpartyXCP/cips/discussions/109#discussioncomment-7245038) by adding a bit more normalization to the issuances data, then the [asset_conservation](https://github.com/CounterpartyXCP/counterparty-lib/blob/e65172a6700a6a45207120ca082da2673d8b240c/counterpartylib/lib/check.py#L152) would only need to check with issuances in the described `issuances_quantity` table. Thus, 0 locked issuances will have no impact because these would have no rows in this table.

**Reply by adamkrellenstein** [reply]
Okay, gotcha. Yeah, I mean I think that fully justifies the existence of zero-quantity locked assets.

Re: normalization of issuances data, my instinct is to go the other way. See my comment on that thread: https://github.com/CounterpartyXCP/cips/discussions/109#discussioncomment-8087724
